src/components/library-app-2/book-form.js

```diff
@@ -1,218 +1,281 @@
+      }
+
+      setIsSubmitting(false);
+    }
+  }, [isSubmitting]);
+
+  const validateField = key => {
+    const field = formConfig[key];
+    if (field.isReq) {
+      const isEmpty = values[key].toString().trim() === '';
+      return isEmpty ? `Enter ${field.label} ` : '';
+    }
+    return '';
+  };
+
+  const handleSubmit = () => {
+    let errors = {};
+
+    formConfigArr.forEach(({key}) => {
+      errors[key] = validateField(key);
+    });
+
+    setFormState(prevFormState => ({
+      ...prevFormState,
+      errors,
+    }));
+    setIsSubmitting(true);
+  };
+
+  const handleFieldChange = (field, value) => {
+    const hasError = !!errors[field];
+
+    setFormState(({values: prevValues, errors: prevErrors}) => ({
+      errors: {...prevErrors, ...(hasError && {[field]: ''})},
+      values: {...prevValues, [field]: value},
+    }));
+  };
+
+  const renderFields = () => {
+    return formConfigArr.map(
+      ({label, key, keyboardType, type, data, defaultButtonText}) => {
+        if (type === 'text') {
+          return (
+            <InputText
+              key={key}
+              styles={{
+                inputContainer: formGroup,
+                label: labelS,
+                input: inputS,
+                error: errorS,
+              }}
+              id={key}
+              onChangeText={handleFieldChange}
+              value={values[key]}
+              error={errors[key]}
+              placeholder={label}
+              keyboardType={keyboardType}
+              label={
+                <>
+                  {label} <Text style={red}>*</Text>{' '}
+                </>
+              }
+            />
+          );
+        } else if (type === 'checkbox') {
+          return (
+            <InputCheckbox
+              key={key}
+              styles={{checkboxContainer, checkbox, label: labelS}}
+              label={label}
+              id={key}
+              onValueChange={handleFieldChange}
+              value={values[key]}
+              error={errors[key]}
+            />
+          );
+        } else if (type === 'dropdown') {
+          return (
+            <SelectDropdown
+              key={key}
+              data={data}
+              styles={{
+                dropdownContainer: formGroup,
+                label: labelS,
+                btn: dropdownBtn,
+                btnTxt: dropdownBtnTxt,
+                row: dropdownRow,
+                rowTxt: dropdownRowTxt,
+              }}
+              onSelect={handleFieldChange}
+              id={key}
+              label={label}
+              ref={dropDownRef}
+              defaultButtonText={defaultButtonText}
+              renderBtnTextAfterSelection={renderItem}
+              renderRowTextForSelection={renderItem}
+            />
+          );
+        }
+      },
+    );
+  };
+
+  return (
+    <SafeAreaView style={container}>
+      <KeyboardAvoidingView
+        behavior={Platform.OS === 'ios' ? 'padding' : null}
```

Comment:
defined these constant props outside component. Any Constant Value or Any Condition Checks if u can move it outside component. Then Do it

Reply:
did this in further exercises
got this comment in later exercises so did it from there